services/sidecar/src/simcore_service_sidecar/core.py

```diff
@@ -28,25 +30,34 @@
 node_port_log.setLevel(config.SIDECAR_LOGLEVEL)
 
 
-async def task_required_resources(node_id: str) -> bool:
+async def task_required_resources(node_id: str) -> Union[bool, None]:
     """Checks if the comp_task's image field if it requires to use the GPU"""
-    async with DBContextManager() as db_engine:
-        async with db_engine.acquire() as db_connection:
-            result = await db_connection.execute(
-                query=comp_tasks.select().where(comp_tasks.c.node_id == node_id)
-            )
-            task = await result.fetchone()
-
-            if not task:
-                log.warning("Task for node_id %s was not found", node_id)
-                raise exceptions.TaskNotFound("Could not find a relative task")
-
-            # Image has to following format
-            # {"name": "simcore/services/comp/itis/sleeper", "tag": "1.0.0", "requires_gpu": false, "requires_mpi": false}
-            return {
-                "requires_gpu": task["image"]["requires_gpu"],
-                "requires_mpi": task["image"]["requires_mpi"],
-            }
+    try:
+        async with DBContextManager() as db_engine:
+            async with db_engine.acquire() as db_connection:
+                result = await db_connection.execute(
+                    query=comp_tasks.select().where(comp_tasks.c.node_id == node_id)
+                )
+                task = await result.fetchone()
+
+                if not task:
+                    log.warning("Task for node_id %s was not found", node_id)
+                    raise exceptions.TaskNotFound("Could not find a relative task")
+
+                # Image has to following format
+                # {"name": "simcore/services/comp/itis/sleeper", "tag": "1.0.0", "requires_gpu": false, "requires_mpi": false}
+                return {
```

Comment:
This is a dict and does not correspond to the type ``bool``, annotated as the function returned type